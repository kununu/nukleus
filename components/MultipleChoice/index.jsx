import React, {Component, PropTypes} from 'react';

import styles from './index.scss';

import {
  controlLabel,
  controlLabelRequired,
  controlNote,
  formControl,
  formGroup
} from '../index.scss';


export default class MultipleChoice extends Component {
  static propTypes = {
    choices: PropTypes.array.isRequired,
    heading: PropTypes.string,
    inputStyle: PropTypes.oneOf(['inline', 'buttons']),
    isRequired: PropTypes.bool,
    name: PropTypes.string.isRequired,
    query: PropTypes.object,
    requiredLabel: PropTypes.string
  };

  static defaultProps = {
    heading: '',
    headingStyle: 'control-label',
    inputStyle: 'inline',
    isRequired: false,
    query: {},
    requiredLabel: ''
  };

  state = {
    choices: this.props.choices || []
  };

  componentWillMount () {
    const {query, name} = this.props;
    if (!query[name]) return;
    this.updateValue(this.getChoicesToUpdate(query[name]), 'checked');
  }

  componentWillReceiveProps (nextProps) {
    const {query, name} = nextProps;
    // We do not yet control the state when the query does not change,
    // for instance when updates are triggered by tweaking with other components.
    if (nextProps.query === this.props.query) return;
    if (!query[name]) {
      this.updateValue(this.state.choices, 'unchecked');
    } else {
      this.updateValue(this.getChoicesToUpdate(query[name]), 'checked');
    }
  }

  onChange (choice) {
    this.updateValue([choice]);
  }

  getChoicesToUpdate (newChoices) {
    return this.state.choices.filter(choice => [].concat(newChoices).some(value => value === choice.value));
  }

  updateValue (newChoices, status = 'toggle') {
    this.setState({
      choices: this.state.choices.map(choice => {
        if (newChoices.some(newChoice => newChoice === choice)) {
          /* eslint-disable sorting/sort-object-props */
          const newStatus = {
            checked: true,
            unchecked: false,
            toggle: !choice.isChecked
          }[status];
          /* eslint-enable sorting/sort-object-props */
          return {
            ...choice,
            isChecked: newStatus
          };
        }
        return choice;
      })
    });
  }

  render () {
    const {choices} = this.state;

    return (
      <div className={`${styles[this.props.inputStyle]} ${formGroup}`}>
        {this.props.requiredLabel &&
          <span className={`${controlNote} ${controlLabelRequired}`}>
            {this.props.requiredLabel}
          </span>
        }

        {this.props.heading && <div className={controlLabel}>{this.props.heading}</div>}

        <div className={styles.inputContainer}>
          {choices.map(choice =>
            <div className={`${styles.choice}`} key={choice.id}>
              <input
                className={formControl}
                id={`${this.props.name}${choice.id}`}
                name={this.props.name}
                key={choice.id}
                value={choice.value}
                type="checkbox"
                checked={choice.isChecked}
                required={this.props.isRequired}
                onChange={() => this.onChange(choice)} />

              <label htmlFor={`${this.props.name}${choice.id}`}>{choice.label}</label>
            </div>
          )}
        </div>
      </div>
    );
  }
}
