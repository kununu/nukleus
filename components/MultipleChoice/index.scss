// sass-lint:disable class-name-format

@import '~kununu-theme/scss/base/variables';

.choice {
  position: relative;
  display: block;
  margin-bottom: 10px;

  * {
    box-sizing: border-box;
  }

  input {
    display: block;
    position: absolute;
    margin-left: 4px 0 0;
    opacity: 0;
    z-index: -1;

    + label {
      color: $gray-base-80;
      cursor: pointer;
      display: inline-block;
      max-width: 100%;
      padding-left: 30px;
      text-transform: none;
      font-size: $font-size-base;
      font-weight: normal;
      letter-spacing: -.01em;
      min-height: 20px;
      margin-bottom: 0;

      &:before {
        border-radius: 1px;
        box-shadow: 0 0 0 2px $white, 0 0 0 3px $gray-base-20;
        content: ' ';
        display: inline-block;
        height: 12px;
        margin-bottom: 3px;
        margin-left: - $baseline-grid * 4;
        margin-right: ($baseline-grid * 2) - 2;
        text-align: center;
        vertical-align: middle;
        width: 12px;

        html.lt-ie9 & {
          border: 1px solid $gray-base-10;
        }
      }
    }

    &:checked + label {
      font-weight: $font-weight-semibold;

      &:before {
        background-image: url('data:image/jpeg;base64,/9j/4QAYRXhpZgAASUkqAAgAAAAAAAAAAAAAAP/sABFEdWNreQABAAQAAABaAAD/4QMxaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wLwA8P3hwYWNrZXQgYmVnaW49Iu+7vyIgaWQ9Ilc1TTBNcENlaGlIenJlU3pOVGN6a2M5ZCI/PiA8eDp4bXBtZXRhIHhtbG5zOng9ImFkb2JlOm5zOm1ldGEvIiB4OnhtcHRrPSJBZG9iZSBYTVAgQ29yZSA1LjYtYzA2NyA3OS4xNTc3NDcsIDIwMTUvMDMvMzAtMjM6NDA6NDIgICAgICAgICI+IDxyZGY6UkRGIHhtbG5zOnJkZj0iaHR0cDovL3d3dy53My5vcmcvMTk5OS8wMi8yMi1yZGYtc3ludGF4LW5zIyI+IDxyZGY6RGVzY3JpcHRpb24gcmRmOmFib3V0PSIiIHhtbG5zOnhtcD0iaHR0cDovL25zLmFkb2JlLmNvbS94YXAvMS4wLyIgeG1sbnM6eG1wTU09Imh0dHA6Ly9ucy5hZG9iZS5jb20veGFwLzEuMC9tbS8iIHhtbG5zOnN0UmVmPSJodHRwOi8vbnMuYWRvYmUuY29tL3hhcC8xLjAvc1R5cGUvUmVzb3VyY2VSZWYjIiB4bXA6Q3JlYXRvclRvb2w9IkFkb2JlIFBob3Rvc2hvcCBDQyAyMDE1IChNYWNpbnRvc2gpIiB4bXBNTTpJbnN0YW5jZUlEPSJ4bXAuaWlkOjcyMTE1MUY5REZGODExRTZCRUY2QjgyOUE2QkE4RTlEIiB4bXBNTTpEb2N1bWVudElEPSJ4bXAuZGlkOjcyMTE1MUZBREZGODExRTZCRUY2QjgyOUE2QkE4RTlEIj4gPHhtcE1NOkRlcml2ZWRGcm9tIHN0UmVmOmluc3RhbmNlSUQ9InhtcC5paWQ6NzIxMTUxRjdERkY4MTFFNkJFRjZCODI5QTZCQThFOUQiIHN0UmVmOmRvY3VtZW50SUQ9InhtcC5kaWQ6NzIxMTUxRjhERkY4MTFFNkJFRjZCODI5QTZCQThFOUQiLz4gPC9yZGY6RGVzY3JpcHRpb24+IDwvcmRmOlJERj4gPC94OnhtcG1ldGE+IDw/eHBhY2tldCBlbmQ9InIiPz7/7gAmQWRvYmUAZMAAAAABAwAVBAMGCg0AAAVWAAAF6gAABoEAAAcs/9sAhAABAQEBAQEBAQEBAgEBAQICAgEBAgICAgICAgICAwIDAwMDAgMDBAQEBAQDBQUFBQUFBwcHBwcICAgICAgICAgIAQEBAQICAgUDAwUHBQQFBwgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAj/wgARCAAQABYDAREAAhEBAxEB/8QAtgAAAwEAAAAAAAAAAAAAAAAABgcICQEAAQUBAAAAAAAAAAAAAAAAAgMEBQYHCBAAAQQDAQEBAAAAAAAAAAAABgECAwQAEQUSEBMRAAEDAwMDBQEAAAAAAAAAAAERAgMABAVREhMhQWExcbHBQlISAAEDAQUJAAAAAAAAAAAAAAEAEQMTICFBUQQxcaGxweESIjITAQACAgEEAgEFAAAAAAAAAAERIQAxQfBRYXGBoZEQscHR8f/aAAwDAQACEQMRAAAB2rzZAcgAQOGtGFfFRmuBaPXsl//aAAgBAQABBQJhfxX9guLo+JHQqd+MdNAxXqJCE3Vl8M8f/9oACAECAAEFAmFdRbRUVNptpVbzaBeI7wWFXWXeU1//2gAIAQMAAQUC/RNySaxEXUkeRx7+f//aAAgBAgIGPwKg9+eD5KnHfIeCl1BJYsd58h7IzwbcR1Cr6j559ky//9oACAEDAgY/ArX/2gAIAQEBBj8CdhhMRMDtFx04jJ/C60bKyIkysg9xCD3PnQVnM0+5ljhueKSJvUvkkbcxv5vCJ61LmMRGshV17Zt79y9v2KGXzQcbVVjheu6d2rl/PzXHtHGibE6JpX//2gAIAQEDAT8hlQm4DsIn6ahanPHZUOo+hbW5OhStEHMK8vRiD0DrehOefkMvInMM+TboZ+Aja4pqIz//2gAIAQIDAT8hT+L/ACHf6ms6ALt89j5a2RFGbkE9Iu+fRiMXcfR7Pky3l5B2+70n1nEV2z//2gAIAQMDAT8h0M0DeSbnOZsP6f/aAAwDAQACEQMRAAAQm19//9oACAEBAwE/EGHAkQmc4NFAIJia1OzaJaRYy/0DA6FabgJi7rMrQ4n0BC1bOWyLfckZAcvM1WVQubXtPU6Ap2EREZ//2gAIAQIDAT8QCjAwQTDCJTahSVCsSiQF7A6Dlbf6BiHDslMMcgnhStDkgFSBa7QOeS33JEYC2Ha5btubf5cerFCIiIjURUajP//aAAgBAwMBPxAfu4B/wYSTT/ZeSfeYq4/3yCM//9k=');
        background-repeat: no-repeat;
        background-position: center;
        background-size: 100%;
      }
    }
  }
}

.inline {
  position: relative;

  @media (min-width: $screen-xs-min) {
    display: flex;
    flex-flow: row;
    align-items: base;
  }

  @media (min-width: $screen-sm-min) {
    padding-top: 6px;
  }

  > div:first-child {
    margin-right: 10px;
    width: 140px;
    max-width: 100%;
    display: inline-block;
  }

  .choice:first-child {
    margin-top: 0;
  }

  .choice:last-child {
    margin-bottom: 0;
  }

  &.paddingTop {
    @media (min-width: $screen-sm-min) {
      padding-top: 25px;
    }
  }
}

form .inline {
  > div:first-child {
    color: $gray-base-60;
    text-transform: uppercase;
    font-weight: 300;
    font-size: 11px;
    min-width: 60px;
  }
}

.inputContainer {
  position: relative;
  width: 100%;
}

.buttons {
  .inputContainer {
    display: block;
    width: auto;
    margin-left: -6px;
    margin-right: -6px;

    @supports (display: flex) {
      display: flex;
      flex-wrap: wrap;
      flex-direction: row;
    }
  }

  .choice {
    display: inline-block;
    padding: 5px;

    &,
    & + .choice {
      margin: 0;
    }

    @supports (display: flex) {
      flex-shrink: 0;
      flex-grow: 1;
    }

    input {
      & + label {
        color: $blue;
        border: 1px solid $blue;
        transition: background-color .25s;
        font-weight: $font-weight-regular;

        @media (max-width: $screen-sm-max) {
          font-size: $font-size-xs;
        }
      }

      & + label:hover,
      &:focus + label {
        background-color: transparent;
        color: $blue;
      }

      &:checked + label,
      &:focus + label,
      &:visited + label {
        outline: 0;
      }

      &:active + label {
        background-color: $blue;
        color: $white;
      }

      &:checked + label {
        background-color: $blue;
        color: $white;

        &,
        &:hover {
          background-color: $blue;
          color: $white;
        }
      }

      & + label,
      &:checked label {
        background-color: transparent;
        border-radius: $border-radius-base;
        padding-left: 0;
        padding: 6px;
        text-align: center;
        transition: background-color .25s;

        @supports (display: flex) {
          width: 100%;
        }
      }

      & + label:before,
      &:checked + label:before {
        display: none;
      }
    }
  }
}
