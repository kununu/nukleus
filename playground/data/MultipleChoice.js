export const multipleChoiceTwoItems = [
  {
    id: 'option-1',
    isChecked: true,
    label: 'flex. Arbeitszeit',
    value: 'option-1'
  },
  {
    id: 'option-2',
    isChecked: false,
    label: 'Homeoffice',
    value: 'option-2'
  }
];

export const multipleChoiceManyItems = [
  {
    id: 'option-1',
    isChecked: true,
    label: 'flex. Arbeitszeit',
    value: 'option-1'
  },
  {
    id: 'option-2',
    isChecked: false,
    label: 'Homeoffice',
    value: 'option-2'
  },
  {
    id: 'option-3',
    isChecked: false,
    label: 'Kantine',
    value: 'option-2'
  },
  {
    id: 'option-4',
    isChecked: false,
    label: 'Essenszulagen',
    value: 'option-2'
  },
  {
    id: 'option-5',
    isChecked: false,
    label: 'Kinderbetreuung',
    value: 'option-2'
  },
  {
    id: 'option-6',
    isChecked: false,
    label: 'Betr. Altersvorsorge',
    value: 'option-2'
  },
  {
    id: 'option-7',
    isChecked: false,
    label: 'Barrierefreiheit',
    value: 'option-2'
  },
  {
    id: 'option-8',
    isChecked: false,
    label: 'Gesundheitsma√ünahmen',
    value: 'option-2'
  },
  {
    id: 'option-9',
    isChecked: false,
    label: 'Betriebsartzt',
    value: 'option-4'
  },
  {
    id: 'option-10',
    isChecked: false,
    label: 'Coaching',
    value: 'option-4'
  },
  {
    id: 'option-11',
    isChecked: false,
    label: 'Parkplatz',
    value: 'option-4'
  },
  {
    id: 'option-12',
    isChecked: false,
    label: 'gute Verkehrsanbindung',
    value: 'option-4'
  },
  {
    id: 'option-13',
    isChecked: false,
    label: 'Mitarbeiterrabatte',
    value: 'option-4'
  },
  {
    id: 'option-14',
    isChecked: false,
    label: 'Firmenwagen',
    value: 'option-4'
  },
  {
    id: 'option-15',
    isChecked: false,
    label: 'Mitarbeiterhandy',
    value: 'option-4'
  },
  {
    id: 'option-16',
    isChecked: false,
    label: 'Mitarbeiterbeteiligung',
    value: 'option-4'
  },
  {
    id: 'option-17',
    isChecked: false,
    label: 'Mitarbeiterevents',
    value: 'option-4'
  },
  {
    id: 'option-18',
    isChecked: false,
    label: 'Internetnutzung',
    value: 'option-4'
  },
  {
    id: 'option-19',
    isChecked: false,
    label: 'Hunde geduldet',
    value: 'option-4'
  }
];

export default multipleChoiceManyItems;
