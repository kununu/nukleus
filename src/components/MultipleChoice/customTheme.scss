// sass-lint:disable class-name-format

@import '~@kununu/kununu-theme-v2/scss/base/variables';

.buttons {
  .choiceInnerContainer {
    display: block;
    width: auto;
    margin-left: -6px;
    margin-right: -6px;

    @supports (display: flex) {
      display: flex;
      flex-wrap: wrap;
      flex-direction: row;
    }
  }

  .choice {
    display: inline-block;
    padding: 5px;

    &,
    & + .choice {
      margin: 0;
    }

    @supports (display: flex) {
      flex-shrink: 0;
      flex-grow: 1;
    }
  }

  .choice input {
    display: block;
    position: absolute;
    margin-left: 4px;
    opacity: 0;
    z-index: -1;

    & + label {
      color: $green;
      border: 2px solid $green;
      transition: background-color .25s;
      font-weight: $font-weight-semibold;
      border-radius: 20px;

      @media (max-width: $screen-sm-max) {
        font-size: $font-size-xs;
      }
    }

    & + label:hover,
    &:focus + label {
      background-color: transparent;
      color: $green;
    }

    &:checked + label,
    &:focus + label,
    &:visited + label {
      outline: 0;
    }

    &:active + label {
      background-color: $green;
      color: $white;
    }

    &:checked + label {
      background-color: $green;
      color: $white;

      &,
      &:hover {
        background-color: $green;
        color: $white;
      }
    }

    & + label,
    &:checked label {
      background-color: transparent;
      border-radius: 20px;
      padding: 8px;
      text-align: center;
      transition: background-color .25s;

      @supports (display: flex) {
        width: 100%;
      }
    }

    & + label:before,
    &:checked + label:before {
      display: none;
    }
  }
}
