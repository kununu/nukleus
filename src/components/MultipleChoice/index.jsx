import React from 'react';
import PropTypes from 'prop-types';

import ThemeContext from 'utils/themeContext';
import themeable from 'utils/theming';

import Error from '../Error';
import Label from '../Label';
import sharedStyles from '../index.module.scss';

import styles from './index.module.scss';

export default class MultipleChoice extends React.Component {
  static propTypes = {
    choices: PropTypes.arrayOf(PropTypes.shape({ // eslint-disable-line react/no-unused-prop-types
      id: PropTypes.string,
      isChecked: PropTypes.bool,
      label: PropTypes.string,
    })),
    disabled: PropTypes.oneOf(['none', 'all', 'uncheckedOnly', 'checkedOnly']),
    error: PropTypes.string,
    errorSubInfo: PropTypes.string,
    heading: PropTypes.string,
    inputStyle: PropTypes.oneOf(['inline', 'buttons']),
    isRequired: PropTypes.bool,
    label: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ]),
    labelHidden: PropTypes.bool,
    name: PropTypes.string.isRequired,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    options: PropTypes.arrayOf(PropTypes.shape({ // eslint-disable-line react/no-unused-prop-types
      id: PropTypes.string,
      isChecked: PropTypes.bool,
      label: PropTypes.string,
    })),
    query: PropTypes.object, // eslint-disable-line react/forbid-prop-types
    reference: PropTypes.func,
    requiredLabel: PropTypes.string,
  };

  static defaultProps = {
    choices: [],
    disabled: 'none',
    error: null,
    errorSubInfo: null,
    heading: null,
    inputStyle: 'inline',
    isRequired: false,
    label: null,
    labelHidden: false,
    onBlur: () => {},
    onChange: () => {},
    onFocus: () => {},
    options: [],
    query: {},
    reference: () => {},
    requiredLabel: '',
  };

  state = {
    choices: MultipleChoice.getCorrectChoiceValues(this.props),
    showError: false,
  };

  componentWillMount () {
    // Show error, if already set
    const {query, name, error} = this.props;

    if (error !== null) this.showError();

    if (!query[name]) return;
    this.updateValue(this.getChoicesToUpdate(query[name]), 'checked');
  }

  componentWillReceiveProps (nextProps) {
    if (nextProps.error) this.showError();
    const newChoiceValues = MultipleChoice.getCorrectChoiceValues(nextProps);
    const oldChoiceValues = MultipleChoice.getCorrectChoiceValues(this.props);

    const {query} = this.props;
    const {choices} = this.state;

    // If new options props are passed to the component
    // We replace the current options with the new ones.
    if (JSON.stringify(newChoiceValues) !== JSON.stringify(oldChoiceValues)) {
      this.setState({
        choices: newChoiceValues,
      });
    }

    // We do not yet control the state when the query does not change,
    // for instance when updates are triggered by tweaking with other components.
    if (nextProps.query === query) return;

    if (!nextProps.query[nextProps.name]) {
      this.updateValue(choices, 'unchecked');
    } else {
      this.updateValue(this.getChoicesToUpdate(nextProps.query[nextProps.name]), 'checked');
    }
  }

  onChange (choice) {
    const {onChange} = this.props;

    this.updateValue([choice], 'toggle', () => {
      const {choices} = this.state;

      onChange(choice, choices);
    });
  }

  get containerClassNames () {
    const {inputStyle, requiredLabel} = this.props;

    const inputStyles = inputStyle.split(' ');

    const classNames = ['formGroup', 'choiceContainer', ...inputStyles];

    if (requiredLabel) classNames.push('paddingTop');

    return classNames;
  }

  static getCorrectChoiceValues ({options, choices}) {
    return (options.length && options) || choices;
  }

  getChoicesToUpdate (newChoices) {
    const {choices} = this.state;

    return choices.filter(choice => [].concat(newChoices).some(value => value === choice.value));
  }

  updateValue (newChoices, status, cb = () => {}) {
    const {choices} = this.state;

    // cb get's fired when setState is finished
    this.setState({
      choices: choices.map((choice) => {
        if (newChoices.some(newChoice => newChoice === choice)) {
          const newStatus = {
            checked: true,
            unchecked: false,
            toggle: !choice.isChecked,
          }[status];

          return {
            ...choice,
            isChecked: newStatus,
          };
        }
        return choice;
      }),
    }, cb);
  }

  showError () {
    this.setState({showError: true});
  }

  hideError () {
    this.setState({showError: false});
  }

  hasError () {
    const {error} = this.props;
    const {showError} = this.state;

    return showError && error;
  }

  isDisabled (checked) {
    const {disabled} = this.props;

    if (disabled === 'all' || (disabled === 'uncheckedOnly' && !checked) || (disabled === 'checkedOnly' && checked)) return true;
    return false;
  }

  render () {
    const {choices} = this.state;
    const {
      error,
      errorSubInfo,
      isRequired,
      inputStyle,
      label,
      labelHidden,
      heading,
      name,
      onBlur,
      onFocus,
      reference,
      requiredLabel,
    } = this.props;

    const labelStyles = inputStyle === 'inline' ? 'inlineLabel' : '';

    return (
      <ThemeContext.Consumer>
        {(context) => {
          const theme = themeable({...sharedStyles, ...styles, ...context});

          return (
            <div
              className={theme(...this.containerClassNames)}
              id={`${name}-container`}
            >
              {requiredLabel && (
              <span className={theme('controlNote', 'controlLabelRequired')}>
                {requiredLabel}
              </span>
              )}

              {label || heading ? (
                <Label
                  classNames={theme(labelStyles)}
                  isTitle
                  labelHidden={labelHidden}
                  value={label || heading}
                />
              ) : null }

              <div className={theme('choiceInnerContainer')}>
                {choices.map(choice => (
                  <div
                    className={theme('choice')}
                    key={choice.id}
                  >
                    <input
                      checked={choice.isChecked}
                      className={theme('formControl')}
                      id={`${name}${choice.id}`}
                      key={choice.id}
                      name={name}
                      onBlur={onBlur}
                      onChange={() => this.onChange(choice)}
                      onFocus={onFocus}
                      ref={reference}
                      required={isRequired}
                      type="checkbox"
                      value={choice.value}
                      disabled={this.isDisabled(choice.isChecked)}
                    />

                    <label htmlFor={`${name}${choice.id}`}>{choice.label}</label>
                  </div>
                ))}
              </div>
              {this.hasError() && (
              <Error
                info={error}
                subInfo={errorSubInfo}
              />
              )}
            </div>
          );
        }}
      </ThemeContext.Consumer>
    );
  }
}
