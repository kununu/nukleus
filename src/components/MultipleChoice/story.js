import React from 'react';
import {storiesOf} from '@storybook/react';
import {defaultOptions} from 'defaultOptions';
import {withOptions} from '@storybook/addon-options';
// import {action} from '@storybook/addon-actions';
// import {withNotes} from '@storybook/addon-notes';
// import {withInfo} from '@storybook/addon-info';
// import {withKnobs, text, boolean} from '@storybook/addon-knobs/react';
// import backgrounds from '@storybook/addon-backgrounds';

import MultipleChoice from './index';

storiesOf('MultipleChoice', module)
  .addDecorator(withOptions({...defaultOptions}))
  .add('basics', () => (
    <MultipleChoice
      name="choices[]"
      label="MultipleChoice"
      options={
      [
        {
          id: 'option-a',
          isChecked: true,
          label: 'option a',
          value: 'option-a'
        },
        {
          id: 'option-b',
          isChecked: false,
          label: 'option b',
          value: 'option-b'
        }]} />
  ))
  .add('Button style', () => (
    <MultipleChoice
      label="Multiple choice component"
      name="choices[]"
      inputStyle="buttons"
      options={
      [{
        id: 'option-1',
        isChecked: true,
        label: 'Elephant',
        value: 'option-1'
      },
      {
        id: 'option-2',
        isChecked: false,
        label: 'Fox',
        value: 'option-2'
      },
      {
        id: 'option-3',
        isChecked: false,
        label: 'Dog',
        value: 'option-3'
      },
      {
        id: 'option-4',
        isChecked: false,
        label: 'Dasypus novemcinctus',
        value: 'option-4'
      },
      {
        id: 'option-5',
        isChecked: false,
        label: 'Horse',
        value: 'option-5'
      },
      {
        id: 'option-6',
        isChecked: false,
        label: 'Chimpanzee',
        value: 'option-6'
      },
      {
        id: 'option-7',
        isChecked: false,
        label: 'Goldfish',
        value: 'option-7'
      }]} />
  ));
