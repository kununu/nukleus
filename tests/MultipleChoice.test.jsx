import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import MultipleChoice from 'MultipleChoice'; // eslint-disable-line import/no-unresolved, import/extensions, import/no-extraneous-dependencies

const choice = (
  <MultipleChoice
    name="choice[]"
    heading="Test"
    isRequired
    choices={
    [
      {
        id: 'option-1',
        isChecked: false,
        label: 'test',
        value: 'test'
      }
    ]
  } />
);

const choices = (
  <MultipleChoice
    name="choice[]"
    heading="MultipleChoice"
    choices={
    [{
      id: 'option-1',
      isChecked: false,
      label: 'option 1',
      value: 'option-1'
    },
    {
      id: 'option-2',
      isChecked: false,
      label: 'option 2',
      value: 'option-2'
    },
    {
      id: 'option-3',
      isChecked: false,
      label: 'option 3',
      value: 'option-2'
    },
    {
      id: 'option-4',
      isChecked: false,
      label: 'option 4',
      value: 'option-4'
    }]} />
);

test('Renders choice without crashing', () => {
  const component = renderer.create(choice);
  expect(component.toJSON()).toMatchSnapshot();
});

test('Changes status on choice change', () => {
  const component = mount(choice);
  component.find('input').simulate('change');
  expect(toJson(component)).toMatchSnapshot();
});

test('Renders choices without crashing', () => {
  const component = renderer.create(choices);
  expect(component.toJSON()).toMatchSnapshot();
});

test('Change status of choices change', () => {
  const component = mount(choices);
  component.find({value: 'option-1'}).simulate('change');
  component.find({value: 'option-4'}).simulate('change');
  expect(toJson(component)).toMatchSnapshot();
});
