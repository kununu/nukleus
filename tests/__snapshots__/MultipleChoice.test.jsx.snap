exports[`test Change status of choices change 1`] = `
<MultipleChoice
  choices={
    Array [
      Object {
        "id": "option-1",
        "isChecked": false,
        "label": "option 1",
        "value": "option-1",
      },
      Object {
        "id": "option-2",
        "isChecked": false,
        "label": "option 2",
        "value": "option-2",
      },
      Object {
        "id": "option-3",
        "isChecked": false,
        "label": "option 3",
        "value": "option-2",
      },
      Object {
        "id": "option-4",
        "isChecked": false,
        "label": "option 4",
        "value": "option-4",
      },
    ]
  }
  heading="MultipleChoice"
  headingStyle="control-label"
  inputStyle="inline"
  isRequired={false}
  name="choice[]"
  query={Object {}}
  requiredLabel="">
  <div
    className="inline formGroup ">
    <div
      className="controlLabel">
      MultipleChoice
    </div>
    <div
      className="inputContainer">
      <div
        className="choice">
        <input
          checked={true}
          className="formControl"
          id="choice[]option-1"
          name="choice[]"
          onChange={[Function]}
          required={false}
          type="checkbox"
          value="option-1" />
        <label
          htmlFor="choice[]option-1">
          option 1
        </label>
      </div>
      <div
        className="choice">
        <input
          checked={false}
          className="formControl"
          id="choice[]option-2"
          name="choice[]"
          onChange={[Function]}
          required={false}
          type="checkbox"
          value="option-2" />
        <label
          htmlFor="choice[]option-2">
          option 2
        </label>
      </div>
      <div
        className="choice">
        <input
          checked={false}
          className="formControl"
          id="choice[]option-3"
          name="choice[]"
          onChange={[Function]}
          required={false}
          type="checkbox"
          value="option-2" />
        <label
          htmlFor="choice[]option-3">
          option 3
        </label>
      </div>
      <div
        className="choice">
        <input
          checked={true}
          className="formControl"
          id="choice[]option-4"
          name="choice[]"
          onChange={[Function]}
          required={false}
          type="checkbox"
          value="option-4" />
        <label
          htmlFor="choice[]option-4">
          option 4
        </label>
      </div>
    </div>
  </div>
</MultipleChoice>
`;

exports[`test Changes status on choice change 1`] = `
<MultipleChoice
  choices={
    Array [
      Object {
        "id": "option-1",
        "isChecked": false,
        "label": "test",
        "value": "test",
      },
    ]
  }
  heading="Test"
  headingStyle="control-label"
  inputStyle="inline"
  isRequired={true}
  name="choice[]"
  query={Object {}}
  requiredLabel="">
  <div
    className="inline formGroup ">
    <div
      className="controlLabel">
      Test
    </div>
    <div
      className="inputContainer">
      <div
        className="choice">
        <input
          checked={true}
          className="formControl"
          id="choice[]option-1"
          name="choice[]"
          onChange={[Function]}
          required={true}
          type="checkbox"
          value="test" />
        <label
          htmlFor="choice[]option-1">
          test
        </label>
      </div>
    </div>
  </div>
</MultipleChoice>
`;

exports[`test Renders choice without crashing 1`] = `
<div
  className="inline formGroup ">
  
  <div
    className="controlLabel">
    Test
  </div>
  <div
    className="inputContainer">
    <div
      className="choice">
      <input
        checked={false}
        className="formControl"
        id="choice[]option-1"
        name="choice[]"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value="test" />
      <label
        htmlFor="choice[]option-1">
        test
      </label>
    </div>
  </div>
</div>
`;

exports[`test Renders choices without crashing 1`] = `
<div
  className="inline formGroup ">
  
  <div
    className="controlLabel">
    MultipleChoice
  </div>
  <div
    className="inputContainer">
    <div
      className="choice">
      <input
        checked={false}
        className="formControl"
        id="choice[]option-1"
        name="choice[]"
        onChange={[Function]}
        required={false}
        type="checkbox"
        value="option-1" />
      <label
        htmlFor="choice[]option-1">
        option 1
      </label>
    </div>
    <div
      className="choice">
      <input
        checked={false}
        className="formControl"
        id="choice[]option-2"
        name="choice[]"
        onChange={[Function]}
        required={false}
        type="checkbox"
        value="option-2" />
      <label
        htmlFor="choice[]option-2">
        option 2
      </label>
    </div>
    <div
      className="choice">
      <input
        checked={false}
        className="formControl"
        id="choice[]option-3"
        name="choice[]"
        onChange={[Function]}
        required={false}
        type="checkbox"
        value="option-2" />
      <label
        htmlFor="choice[]option-3">
        option 3
      </label>
    </div>
    <div
      className="choice">
      <input
        checked={false}
        className="formControl"
        id="choice[]option-4"
        name="choice[]"
        onChange={[Function]}
        required={false}
        type="checkbox"
        value="option-4" />
      <label
        htmlFor="choice[]option-4">
        option 4
      </label>
    </div>
  </div>
</div>
`;
